### PR TITLE
[App Configuration] Not log value of key

### PR DESCRIPTION
**Pull request by ivywei0125**
# Description
When content type is json, we'll check if the value is a valid json. If invalid, throw `JsonProcessingException`. 

Before fix, for a key-value: `test_key: test_value`, the error message in the log is as the following, contains the value "test_value".
_"com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'test_value_3': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"test_value_3"; line: 1, column: 13]"_

We shouldn't include the value in logs to avoid any security issue. 
After fix, the error message is:
_"java.io.IOException: Invalid value of key '/foo/test_key_3'. Expected type: JSON String, Number, Array, Object or token 'null', 'true' or 'false'"_

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java [hunk 1]
```diff
@@ -44,10 +45,16 @@ static boolean isJsonContentType(String contentType) {
...
-        throws JsonProcessingException {
+        throws IOException {
         Map<String, Object> settings = new HashMap<>();
-        JsonNode json = MAPPER.readTree(setting.getValue());
-        parseSetting(setting.getKey(), json, settings);
+        try {
+            JsonNode json = MAPPER.readTree(setting.getValue());
+            parseSetting(setting.getKey(), json, settings);
+        } catch (JsonProcessingException e) {
+            throw new IOException(
```
Comment (mrm9084) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java:54]: InvalidConfigurationPropertyValueException

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java (hunk 1 above)
Comment (ivywei0125) [reply sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java:54]: I checked the constructor of `InvalidConfigurationPropertyValueException`. It requires `name`, `value` and `reason`. 
What shall we provide for `value`? We didn't want to log the value

Comment (mrm9084) [reply sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java:54]: If you read the description a null value for the `value` field is allowed.

Comment (ivywei0125) [reply sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java:54]: But is it a little wired that if we use a `null` value? 
The error message would be `Property test_key with value 'null' is invalid: Expected type: JSON String, Number, Array, Object or token 'null', 'true' or 'false'`

Besides, I also tried to pass an empty string as value. It's also a little wired.
`Property test_key with value '' is invalid: Expected type: JSON String, Number, Array, Object or token 'null', 'true' or 'false'`

Comment (mrm9084) [reply sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java:54]: Maybe we do what we do with our logs and put `<Redacted>`

Comment (ivywei0125) [reply sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/JsonConfigurationParser.java:54]: Do you mean we still use `null` when throw `InvalidConfigurationPropertyValueException`. But when logging this error, we replace the `null` string to other string?

**Review by mrm9084 — APPROVED** [review]

**Pull request merged**